flytekit/annotated/testing.py

```diff
@@ -43,25 +44,23 @@ def _log(*args, **kwargs):
     t.execute = _captured_fn
 
 
-def patch(target: Union[PythonTask, Workflow]):
-    if not isinstance(target, PythonTask) and not isinstance(target, Workflow):
+def patch(target: Union[PythonTask, Workflow, ReferenceEntity]):
+    if (
+        not isinstance(target, PythonTask)
+        and not isinstance(target, Workflow)
+        and not isinstance(target, ReferenceEntity)
+    ):
+
         raise Exception("Can only use mocks on tasks/workflows declared in Python.")
 
     def wrapper(test_fn):
         def new_test(*args, **kwargs):
             logger.warning(f"Invoking mock method for target: '{target.name}'")
             m = MagicMock()
-            if isinstance(target, PythonTask):
-                saved = target.execute
-                target.execute = m
-            else:
-                saved = target._workflow_function
-                target._workflow_function = m
+            saved = target.execute
+            target.execute = m
```

Comment:
what happened to Workflow?

Reply:
nm, just saw it

Reply:
See comment in pr desc.  I added an execute function to workflow.